Run tests as GitHub action

markuspf opened the pull request:
Quick-and-dirty GitHub Action to run the tests in that are currently run in Travis and AppVeyor, adds gcc 10 and 11 and clang 11 and 12.

Ready for review now.

jsteemann commented on .github/workflows/cmake.yml:
```diff
@@ -0,0 +1,109 @@
+env:
+  CMAKE_BUILD_PARALLEL_LEVEL: 4
+  CXX_STANDARD: 20
+  SANITIZER: false
```
We can get rid of this variable here, as it is always `false`.

jsteemann commented on .github/workflows/cmake.yml:
```diff
@@ -0,0 +1,109 @@
+          - {os: ubuntu-latest, compiler: gcc, version: '10',
+             build_type: Debug, hash_type: xxhash,
+             flags: '-fsanitize=address -fsanitize=leak -fsanitize=undefined -fsanitize-address-use-after-scope -fno-omit-frame-pointer',
+             ld_flags: '-fsanitize=address -fsanitize=leak -fsanitize=undefined -fsanitize=address-use-after-scope -fno-omit-frame-pointer'}
```
It would be good to have some variable set here in case we are building with sanitizers.
Right now the script `scripts/travis-test.sh` filters on whether the `SANITIZER` variable is set to `true` (currently never), and if it were, it would not run the fuzz tests.
I imagine that if we build with sanitizers, we probably want to reduce the amount of fuzz testing to keep the execution times at bay. If execution of the fuzz tests with sanitizers is fast enough, then we don't need to set a variable here, and we can remove the `if` condition from `scripts/travis-test.sh` that check if sanitizers are enabled.

markuspf replied:
Should now set `SANITZER` environment variable, though I do think that the fuzz tests did run on the sanitizer run and there wasn't too much of an issue

jsteemann commented on .github/workflows/cmake.yml:
```diff
@@ -0,0 +1,109 @@
+
+    - name: Test
+      working-directory: ${{github.workspace}}
+      run: scripts/travis-test.sh
```
As we are moving away from Travis CI with this PR, we should probably also rename the script from `scripts/travis-test.sh` to `scripts/execute-tests.sh` or similar.
There are also some other Travis CI-related scripts in the `scripts` directory that we could delete in this PR. They should have become unnecessary.

markuspf replied:
I removed all of travis and renamed the scripts with my latest push.

markuspf commented:
Probably want to update the badges as well, and if we want to remove appveyor, too?

markuspf commented:
I added the badge, though it's unclear to me how to test whether it'll work before the PR  is merged.

I also removed the AppVeyor YAML file now.

I think unless there are other problems with this PR it is ready to merge. Happy to update things if there are problems that arise.

markuspf commented:
Fixed the badge (wrong filename...)
